app/scripts/modules/core/src/pipeline/config/stages/monitorPipeline/MonitorPipelineStageExecutionDetails.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { UISref } from '@uirouter/react';
+
+import { IExecutionDetailsSectionProps, ExecutionDetailsSection } from '../common';
+import { StageFailureMessage } from '../../../details';
+import _ from 'lodash';
+
+export function MonitorPipelineStageExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const {
+    application,
+    execution,
+    stage: { context = {}, outputs = {} },
+    stage,
+    name,
+    current,
+  } = props;
+
+  const statuses = Object.keys(outputs.executionStatuses).map((executionId) => {
+    const status = outputs.executionStatuses[executionId];
+    status.executionId = executionId;
+
+    const generalException: any = status.exception;
+    if (generalException) {
+      const errors = (generalException.details?.errors ?? []).filter((m: any) => !!m);
+
+      if (errors.length) {
+        status.failureMessage = errors.join('\n\n');
+      } else {
+        status.failureMessage = generalException.details?.error ?? null;
+      }
+    }
+
+    return status;
+  });
+
+  return (
+    <ExecutionDetailsSection name={name} current={current}>
+      <div className="row">
+        <div className="col-md-12">
+          <h5>Monitor behavior</h5>
+          {context.monitorBehavior}
+
+          <h5>Executions</h5>
+          {_.sortBy(statuses, 'status').map((status) => {
```

Comment:
probably more idiomatic to sort outside of the JSX.  Way up there ☝️

Reply:
yes, thank you!